app/models/mixins/purging_mixin.rb

```diff
@@ -98,8 +98,40 @@ def purge_by_scope(older_than = nil, window = nil, &block)
       _log.info("Purging #{table_name.humanize}...Complete - Deleted #{total} records")
     end
 
+    def purge_by_orphaned(fk_name, polymorphic, join_table = "", window = purge_window_size)
+      _log.info("Purging orphans in #{table_name.humanize}...")
+      total = polymorphic ? purge_polymorphic_orphans(fk_name, window) : purge_orhans(fk_name, join_table, window)
+      _log.info("Purging orphans in #{table_name.humanize}...Complete - Deleted #{total} records")
+      total
+    end
+
     private
 
+    def purge_polymorphic_orphans(fk_name, window)
+      type_column = "#{fk_name}_type"
+      total = 0
+      polymorphic_classes(type_column).each do |klass|
+        resource_table = connection.quote_table_name(klass.table_name)
+        scope = joins("LEFT OUTER JOIN #{resource_table} ON #{table_name}.resource_id = #{resource_table}.id")
+                .where(resource_table => {:id => nil})
+                .where("#{table_name}.#{connection.quote_column_name(type_column)} = #{connection.quote(klass.name)}")
+        total += purge_in_batches(scope, window)
+      end
+      total
+    end
+
+    def polymorphic_classes(type_column)
```

Comment:
Can we make this variable `resource_type_column`?  Only because type is it's own thing, and it's confusing to read.

Reply:
Or `polymorphic_type_column` ?